### ISSUE TITLE
Clients may not get updated when macro buttons are added/removed/changed.

### ISSUE THREAD

**Issue by Phergus** [RPTools/maptool#1745]
**Describe the bug**
From #368 
Did a little investigation; the following is happening when the "Player Receive Campaign Macros" is on:

Token:
- adding a macro -> no update on the client
- renaming -> no update on the client
- delete a macro -> no update on the client
- clear group -> no update on the client

Campaign
- adding a macro -> no update on the client
- renaming a macro -> instant update on the client
- renaming a group -> no updated on client
- clear group -> no update on the client
- clear panel -> no update on the client

The client is updated as soon as the token is selected or a Campaign macro is edited.

**Expected behavior**
Expect clients to be updated.

**MapTool Info**
- Version: 1.5.14+ (and probably much older)
- Install: New

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: 10

**Additional context**
See discussion in #368

**Comment by Merudo** [comment]
Yeah, the way the macros are updated is considerably broken.

I fixed it a bit for token macros in previous PR but I didn't touch the Campaign / GM macros. Solving the problem will probably requiring adding a few new server commands.

**Comment by aliasmask** [comment]
Yeah, this is been around for awhile. You just have to know that the GM has to edit macro for it to show up for everyone else.
